master/internal/db/postgres_trial_intg_test.go

```diff
@@ -62,3 +69,111 @@ func TestProtoGetTrial(t *testing.T) {
 	require.NoError(t, err, "failed to query trial")
 	require.Equal(t, trResp.WallClockTime, float64(3), "wall clock time is wrong")
 }
+
+// Covers an issue where checkpoint_view returned multiple records per checkpoint
+// due to the LEFT JOIN raw_steps ON total_batches AND trial_id.
+// This is in this file because AddValidationMetrics broke the assumption the join uses.
+// That assumption being that for each trial_id and total_batches that there
+// is at most one unarchived result.
+func TestAddValidationMetricsDupeCheckpoints(t *testing.T) {
+	ctx := context.TODO()
```

Comment:
`context.Background()` is appropriate for tests (at least its documentation says so) and then this wont show up when i grep for `context.TODO()`

Reply:
That is a good point, I really didn't put any thought into doing ```context.TODO()``` vs ```context.Background```